src/Microsoft.AspNetCore.OData/EnableQueryAttribute.cs

```diff
@@ -234,6 +235,26 @@ public override void OnActionExecuted(ActionExecutedContext actionExecutedContex
                 // been applied to the result; make sure it's also successful.
                 ObjectResult responseContent = actionExecutedContext.Result as ObjectResult;
 
+                if (responseContent == null)
+                {
+                    Type createdODataResultType = actionExecutedContext.Result.GetType().GetGenericArguments().Count() > 0 ?
+                        typeof(CreatedODataResult<>).MakeGenericType(actionExecutedContext.Result.GetType().GetGenericArguments()[0]) : null;
+
+                    Type actionResultType = actionExecutedContext.Result.GetType();
+
```

Comment:
You seem to be computing `actionExecutedContext.Result.GetType()` and `GetGenericArguments()` multiple times here. Suggestion to compute them only once:
```suggestion
                    Type actionResultType = actionExecutedContext.Result.GetType();
                    Type[] actionResultGenericArgs = actionResultType.GetGenericArguments();
                    Type createdODataResultType = actionResultGenericArgs.Length > 0 ?
                        typeof(CreatedODataResult<>).MakeGenericType(actionResultGenericArgs[0]) : null;

```

Reply:
+1

Reply:
Fixed